pkg/R/RDD.R

```diff
@@ -46,8 +58,12 @@ setMethod("initialize", "PipelinedRDD", function(.Object, prev, func, jrdd_val)
   .Object@env$isCached <- FALSE
   .Object@env$isCheckpointed <- FALSE
   .Object@env$jrdd_val <- jrdd_val
-   # This tracks if jrdd_val is serialized
-  .Object@env$serialized <- prev@env$serialized
+  if (!is.null(jrdd_val)) {
+     # This tracks the serialization mode for jrdd_val
+    .Object@env$serializedMode <- prev@env$serializedMode
+  }
+  
+  .Object@colNames <- prev@colNames
```

Comment:
`colNames` can not be inherited from prev, if we can not make sure that `func` does keep them not changed.